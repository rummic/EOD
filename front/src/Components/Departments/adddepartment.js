import React, { Component } from "react";
import { Redirect } from "react-router-dom";
import Sidebar from "../Navbar/sidebar";
import Swal from 'sweetalert2';

const token = sessionStorage.getItem("token");

class adduser extends Component {
  constructor(props) {
    super(props);
    this.state = {
    departmentId: 1,
    userId:1,
      name: "",
      role: "",
      users: []

    };
    this.onChange = this.onChange.bind(this);
  }

  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  componentDidMount(){
    document.title = 'Dodawanie działu';
    fetch("https://localhost:44388/api/Users/" + sessionStorage.getItem('id'), {
      method: "GET",
      headers: {
        "Content-Type": "application/json",
        Authorization: `bearer ${token}`
      }
    })
    .then(response => response.json())
    .then(parseJSON => {
      this.setState({
        role: parseJSON.value.role
      });
    });

    fetch("https://localhost:44388/api/Users", {
      method: "GET",
      headers: {
        "Content-Type": "application/json",
        Authorization: `bearer ${token}`
      }
    }).then(response =>
      response.json().then(responseJSON => {
        console.log(responseJSON);
        this.setState({
          users: responseJSON.value || []
        });
      })
    );

  }

  addDepartment() {
    fetch("https://localhost:44388/api/Departments", {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
        Authorization: `bearer ${token}`
      },
      body: JSON.stringify({
        name: this.state.name
      })
    })
      .then(response => response.json())
      .then(parseJSON => {
          console.log(this.state.name)
          this.setState({
            value: parseJSON.value
          });
        if (parseJSON.hasErrors) {
          Swal.fire({
            type: 'error',
            title: 'Błąd',
            text: 'Dział nie został dodany',
          })
        } else {
          fetch(
            "http://localhost:60148/api/Departments?id=" + this.state.value +"&userId=" + this.state.userId ,{
              method: "PATCH",
              headers: {
                "Content-Type": "application/json",
                Authorization: `Bearer ${token}`
              }
            }
          )
            .then(response => response.json())
            .then(parseJSON => {
              if (parseJSON.hasErrors) {
                Swal.fire({
                  type: 'error',
                  title: 'Błąd',
                  text: 'Dział nie został dodany',
                })
              } else {
                Swal.fire({
                  type: 'success',
                  title: 'Sukces',
                  text: 'Dział został dodany pomyślnie',
                })
                this.props.history.push("/departments");
              }
            });
        }
      });
  }

  render() {
    if (!sessionStorage.getItem("token") || this.role === "User" || this.role === "Admin") {
      return <Redirect to={"/home"} />;
    }
    return (
      <div className="UsersetBox">
        <Sidebar history={this.props.history} />
        <div className="UsersetBox-content">
          <div className="UsersetBox-form">
            <div className="UsersetBox-form-content">
              <label>Nazwa :</label>
              <input
                type="text"
                placeholder="Podaj nazwe"
                name="name"
                onChange={this.onChange}
                required
              />
              <label>Wybierz kierownika :</label>
              <select  value={this.state.userId} name="userId" onChange={(e) => this.setState({userId: e.target.value})}>
                {this.state.users.map((item, i) => (
                 
                  <option   key={i}  value={item.id} style = {{display : item.role ==='Admin' ? 'block' : 'none'}} >{item.login} </option>
                  
                ))}
              </select>
            </div>
            <button
              className="UsersetBox-form-button"
              onClick={this.addDepartment.bind(this)}
            >
              Dodaj
            </button>
          </div>
        </div>
      </div>
    );
  }
}

export default adduser;