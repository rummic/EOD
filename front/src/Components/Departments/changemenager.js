import React, { Component } from "react";
import Sidebar from "../Navbar/sidebar";
import { Redirect } from "react-router-dom";
import { Table } from "react-bootstrap";

const token = sessionStorage.getItem("token");

class filedetails extends Component {
  constructor(props) {
    super(props);
    this.state = {
      id: null,
      name: "",
      manager:[]

    };
    this.onChange = this.onChange.bind(this);
  }

  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  componentWillMount() {
    const obj = this.props.location.state;
    this.setState({
      id: obj.id,
      name: obj.name,
      manager: obj.manager,
      users: [],
      userId:1
    });
  }

  componentDidMount(){
    fetch("https://localhost:44388/api/Users/" + sessionStorage.getItem('id'), {
      method: "GET",
      headers: {
        "Content-Type": "application/json",
        Authorization: `bearer ${token}`
      }
    })
    .then(response => response.json())
    .then(parseJSON => {
      this.setState({
        role: parseJSON.value.role
      });
    });

    fetch("https://localhost:44388/api/Users", {
      method: "GET",
      headers: {
        "Content-Type": "application/json",
        Authorization: `bearer ${token}`
      }
    }).then(response =>
      response.json().then(responseJSON => {
        console.log(responseJSON);
        this.setState({
          users: responseJSON.value || []
        });
      })
    );

  }

  changeMenager(){
    fetch(
      "http://localhost:60148/api/Departments?id=" +this.props.location.state.id +"&userId=" + this.state.userId ,{
        method: "PATCH",
        headers: {
          "Content-Type": "application/json",
          Authorization: `Bearer ${token}`
        }
      }
    )
      .then(response => response.json())
      .then(parseJSON => {
        if (parseJSON.hasErrors) {
          alert("Dział nie został dodany");
        } else {
          alert("Dział został dodany");
          this.props.history.push("/departments");
        }
      });
  }


  
  render() {

    if (!sessionStorage.getItem("token")) {
      return <Redirect to={"/home"} />;
    }
    const obj = this.state
    return (
      
      
        <div className="UsersetBox">
          <Sidebar history={this.props.history} />
          <div className="UsersetBox-content">
            <div className="UsersetBox-form">
              <div className="UsersetBox-form-content">
                <div className="UsersetBox">
                  <Table striped bordered hover size="sm">
                    <tbody>
                          <tr>
                            <th>Nazwa działu</th>
                            <th>Kierownik</th>
                            <th>Wybierz innego kierownika</th> 
                          </tr>
                          <tr>
                          <td>{obj.id}{obj.name}</td>
                          <td >
                            {obj.manager.login}
                          </td>
                          <td>
                          <select value={this.state.userId} name="userId" onChange={(e) => this.setState({userId: e.target.value})}>
                            {this.state.users.map((item, i) => (
                 
                            <option style = {{display : item.role !=='Admin' ? 'none' : 'block'}}  key={i}  value={item.id}  >{item.login} </option>
                  
                              ))}
                            </select>
                          </td>
                          </tr>
                    </tbody>
                  </Table>
                  <button
              className="UsersetBox-form-button"
              onClick={this.changeMenager.bind(this)}
            >
              Zamień
            </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      
    );
  }
}

export default filedetails;