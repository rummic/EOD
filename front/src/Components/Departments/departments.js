import React, { Component } from "react";
import { Redirect,Link } from "react-router-dom";
import Sidebar from "../Navbar/sidebar";
import { Breadcrumb, Table } from "react-bootstrap";

const token = sessionStorage.getItem("token");

class departments extends Component {
  constructor(props) {
    super(props);
    this.state = {
      departments:[],
      role: ""
    };
    this.onChange = this.onChange.bind(this);
  }

  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }
  

  componentDidMount() {
      document.title = 'Lista działów';
    fetch("https://localhost:44388/api/Users/" + sessionStorage.getItem('id'), {
        method: "GET",
        headers: {
          "Content-Type": "application/json",
          Authorization: `bearer ${token}`
        }
      })
      .then(response => response.json())
      .then(parseJSON => {
        this.setState({
          role: parseJSON.value.role
        });
      });

    fetch("https://localhost:44388/api/Departments" , {
      method: "GET",
      headers: {
        "Content-Type": "application/json",
        Authorization: `bearer ${token}`
      }
    })
    .then(response => response.json())
    .then(parseJSON => {
      this.setState({
        departments: parseJSON.value || []
      });
    });
    console.log(this.state.departments);

  }

  render() {
    if (!sessionStorage.getItem("token") || this.state.role === "User" || this.state.role === "Admin" ) {
      return <Redirect to={"/home"} />;
    }
    return (
      <div className="UsersetBox">
        <Sidebar history={this.props.history} />
        <div className="UsersetBox-content">
          <div className="UsersetBox-form">
          <Breadcrumb>
              <Breadcrumb.Item href="/index">EOD</Breadcrumb.Item>
              <Breadcrumb.Item active>Lista Użytkowników</Breadcrumb.Item>
            </Breadcrumb>
            <div className="UsersetBox-form-content">
              <div className="UsersetBox">
                <div>
                <button>
                    <Link to={{ pathname: "./adddepartment" }}>
                      Dodaj dział
                    </Link>
                  </button>
                <Table striped bordered hover size="sm">
                    <thead>
                      <tr>
                        <th>Id</th>
                        <th>Nazwa</th>
                      </tr>
                    </thead>
                    {this.state.departments.map((item, i) => (
                      <tbody key={i}>
                        <tr>
                          <td>{item.id}</td>
                          <td>{item.name}</td>
                          <td>
                            <button>
                            <Link
                                to={{ pathname: "./changemenager", state: item }}
                              >
                                Ustawienia
                              </Link>
                            </button>
                          </td>
                        </tr>
                      </tbody>
                    ))}
                  </Table>
                  </div>
                  </div>
                  </div>
                  </div>
                  </div>
                  </div>
    );
  }
}

export default departments;
