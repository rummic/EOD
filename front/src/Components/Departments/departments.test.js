import React from 'react';
import { shallow } from 'enzyme';
import changemenager from './changemenager';
import departments from './departments';
import adddepartment from './adddepartment';

describe('departments', () => {
    test('Rendering departments component', () => {
        const wrapper = shallow(<departments />);
        expect(wrapper).toMatchSnapshot();
    });
    test('Rendering adddepartment component', () => {
        const wrapper = shallow(<adddepartment />);
        expect(wrapper).toMatchSnapshot();
    });
    test('Rendering changemenager component', () => {
        const wrapper = shallow(<changemenager />);
        expect(wrapper).toMatchSnapshot();
    });
});  