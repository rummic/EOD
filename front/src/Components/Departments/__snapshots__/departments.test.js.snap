// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`departments Rendering adddepartment component 1`] = `ShallowWrapper {}`;

exports[`departments Rendering changemenager component 1`] = `ShallowWrapper {}`;

exports[`departments Rendering departments component 1`] = `ShallowWrapper {}`;

exports[`departments Rendering departments component 2`] = `ShallowWrapper {}`;
